src/traces/histogram2d/sample_defaults.js

```diff
@@ -6,24 +6,26 @@
 * LICENSE file in the root directory of this source tree.
 */
 
-
 'use strict';
 
 var Registry = require('../../registry');
+var Lib = require('../../lib');
 
 module.exports = function handleSampleDefaults(traceIn, traceOut, coerce, layout) {
     var x = coerce('x');
     var y = coerce('y');
+    var xlen = Lib.minRowLength(x);
+    var ylen = Lib.minRowLength(y);
 
     // we could try to accept x0 and dx, etc...
     // but that's a pretty weird use case.
     // for now require both x and y explicitly specified.
-    if(!(x && x.length && y && y.length)) {
+    if(!(x && xlen && y && ylen)) {
```

Comment:
you already did this elsewhere... but one of the nice things about your `Lib.minRowLength` is you can just say `if(!(xlen && ylen))` right?

Reply:
done in https://github.com/plotly/plotly.js/pull/3300/commits/627f24b68052e1cda45fea65fcf20901309bf303